packages/chart/src/CdcChartComponent.tsx

```diff
@@ -716,51 +648,13 @@ const CdcChart = ({
     }
   }, [externalFilters]) // eslint-disable-line
 
-  // This will set the bump chart's default scaling type to date-time
-  useEffect(() => {
-    if (['Bump Chart'].includes(config.visualizationType)) {
-      setConfig({
-        ...config,
-        xAxis: {
-          ...config.xAxis,
-          type: 'date-time'
-        }
-      })
-    }
-  }, [config.visualizationType])
-
-  // Generates color palette to pass to child chart component
   useEffect(() => {
-    if (stateData && config.xAxis && config.runtime?.seriesKeys) {
-      const configPalette = ['Paired Bar', 'Deviation Bar'].includes(config.visualizationType)
-        ? config.twoColor.palette
-        : config.palette
-      const allPalettes: Record<string, string[]> = { ...colorPalettes, ...twoColorPalette }
-      let palette = config.customColors || allPalettes[configPalette]
-      let numberOfKeys = config.runtime.seriesKeys.length
-      let newColorScale
-
-      while (numberOfKeys > palette.length) {
-        palette = palette.concat(palette)
-      }
-
-      palette = palette.slice(0, numberOfKeys)
-
-      newColorScale = () =>
-        scaleOrdinal({
-          domain: config.runtime.seriesLabelsAll,
-          range: palette,
-          unknown: null
-        })
-
-      setColorScale(newColorScale)
+    if (config && data) {
+      const colorScale = getColorScale(state.config, stateData)
+      dispatch({ type: 'SET_COLOR_SCALE', payload: colorScale })
       setLoading(false)
```

Comment:
Should this use a dispatch fn, ie SET_LOADING?

Reply:
Fixed!